src/main.ts

```diff
@@ -0,0 +1,128 @@
+const repository = process.env.GITHUB_REPOSITORY;
+const branch = process.env.GITHUB_REF;
+const event_payload =  process.env.GITHUB_EVENT_PATH || jsonPath;
+let event_name = process.env.GITHUB_EVENT_NAME;
+
+let msteams_webhook_url;
+let job_status;
+let event_id;
+let details;
+let account;
+let message;
+let url;
+
+
+async function parse_inputs() {
+    try {
+
+        // do not proceed if jobs was cancelled
+        job_status = core.getInput('job_status');
+        if (job_status.toUpperCase() === "CANCELLED") {
+            console.log("Job was cancelled, no notification will be sent")
+            process.exit(0);
+        }
+
+        msteams_webhook_url = core.getInput('webhook_url');
+        event_id = core.getInput('event_id');
+
+        const event_payload_data_text =  JSON.stringify(require(event_payload));
+        const event_payload_data = JSON.parse(event_payload_data_text);
+
+        if (event_id) {
+            event_name = event_id;
+        }
+
+        switch(event_name){
+            case 'push':
+                account = event_payload_data['pusher']['name'];
+                message = `**Commit to GitHub** by ${account}`;
+                url = event_payload_data['compare'];
+                details = `Comment: ${event_payload_data['head_commit']['message']}`;
+                break;
+            case 'pull_request':
+                message = `**PR submitted to Github**: ${event_payload_data['pull_request']['title']}`;
+                url = event_payload_data['pull_request']['html_url'];
+                details = `Pr for merging ref ${event_payload_data['pull_request']['head']['ref']} to base branch ${event_payload_data['base']['ref']}`;
+                break;
+            case 'issue':
+                message = `**New/updated GitHub issue**: ${event_payload_data['issue']['title']}`;
+                url = event_payload_data['issue']['html_url'];
+                details = `Issue state: ${event_payload_data['issue']['state']}  - assignee: ${event_payload_data['issue']['assignee']}`;
+                break;
+            case 'issue_comment':
+                message = `**A Github issue comment was posted**: ${event_payload_data['comment']['body']}`;
+                url = event_payload_data['issue']['html_url'];
+                details = `Issue state: ${event_payload_data['issue']['state']} - assignee: ${event_payload_data['issue']['assignee']}`;
+                break;
+            default:
+                if (event_name) {
+                    message = event_name.replace(/_/g, ".");
+                } else {
+                    message = "A GitHub Actions event has occurred"
+                }
+                url = `https://github.com/${repository}/actions`;
+        }
+
+        if (!details) {
+            details = core.getInput('details');
+        }
+    } catch (error) {
+        core.setFailed(error.message);
+    }
+}
+
+
+async function notifyTeams() {
```

Comment:
I see most of your name in this module are snake_case - let's keep things consistent?

Reply:
Sorry, because of working on java at the same time, force of habit --> very fair point, will update the name of this function

Reply:
consistency is king so I don't feel strongly beyond that, but I think it is fair to say that snake case is unconventional for typescript.

Reply:
To be honest for typescript I do not feel strongly against snake_case --> in this case I found it more readable because the information we parse from the github payload comes in this format as well.
One thing I did not do in this PR but I was planning to add in a separate one (I was going for smaller PRs, did not achieve this in this case :( ) was to add a linter. I gave the typical eslint a try, configured it and opted for the airbnb style, I saw you are also using this one in the el-b2b-fe-school. Camel case is one of the returned error, and mostly indentation to be honest. I will add a new PR with those changes on top, otherwise it will not be possible to even go through the changed files in the current one.

Reply:
This should be fixed now with refactoring driven by the addition of eslint.